README.adoc

```diff
@@ -1,5 +1,256 @@
+
+The helpers are available in springEL too with # prefix.
+
+=== Self meaning in different situations
+
+The self mean different thinks in the springEL and Handlebars depend on the YAML properties:
+
+- If actorTypeBased is enabled, then #self refers to the actor.
+
+- If we have a `factoryExpression`, then the self is the current iterated element of collection.
+
+- If we don't have a `factoryExpression` and it is not actorTypeBased, then #self refers to the model.
+
+=== How to reach templateParameters
+
+They are not static and have to be accessible to static helper methods which are called from parallel processes. This can cause issues with parallel building. To void
+this issue the `@ContextAccessor` annotation and a `bindContext` method.
+
+Example:
+[source,java]
+----
+@TemplateHelper
+@ContextAccessor
+public class StoredVariableHelper extends StaticMethodValueResolver {
+
+    public static void bindContext(Map<String, ?> context) {
+        ThreadLocalContextHolder.bindContext(context);
+    }
+
+    public static synchronized String getApiPrefixLocal(Object object) {
+        return (String) ThreadLocalContextHolder.getVariable("apiPrefix");
+    }
+
+    public static synchronized Boolean isGenerateOptionalTypes(Object object) {
+        return Boolean.parseBoolean((String) ThreadLocalContextHolder.getVariable("generateOptionalTypes"));
+    }
+
+    public static synchronized Boolean isGeneratePayloadValidator(Object object) {
+        return Boolean.parseBoolean((String) ThreadLocalContextHolder.getVariable("generatePayloadValidator"));
+    }
+}
+
+----
+
+This only need if we used the templateParameters in a java Helper.
+
+=== Checksum generation
+A checksum is a value calculated from data to detect errors during transmission or storage. It ensures data integrity and any mismatches indicate errors in the data.
+
+* The checksum generation is a goal of the generator-maven-plugin.
+
+Example:
+
+[source,xml]
+----
+                    <execution>
+                        <id>execute-ui-checksum-from-file</id>
+                        <phase>generate-test-resources</phase>
+                        <goals>
+                            <goal>checksum</goal>
+                        </goals>
+                        <configuration>
+                            <ui>${basedir}/src/test/resources/MultiPrincipalTest-ui.model</ui>
+                            <destination>${basedir}/target/test-classes/ui/file</destination>
+                        </configuration>
+                    </execution>
+----
+* The [modelName]ProjectGenerateMojo is responsible for generating the files, while the [modelName]ProjectCalculateChecksumMojo is responsible for generating checksums for the files.
+
+* The .generated-files-[actor]-[actor] contains the actorTypeBased checksums. The .generated-files contains the not actorTypeBased checksums
```

Comment:
I'm pretty sure `[actor]-[actor]` is incorrect. There is no point in repeating the same. Guess it should be model-actor.

Reply:
[b8f2188](https://github.com/BlackBeltTechnology/judo-generator-commons/pull/29/commits/b8f218814caf482868a6d80c23e83c6ca810567d)